src/createBaseForm.js

```diff
@@ -168,6 +170,13 @@ function createBaseForm(option = {}, mixins = []) {
       getFieldDecorator(name, fieldOption) {
         const props = this.getFieldProps(name, fieldOption);
         return (fieldElem) => {
+          warning(
+            !this.elementRendered[name],
+            `Duplicate field names will result ` +
+            `in both fields getting edited together. ` +
+            `Field names must be unique`
+          )
+          this.elementRendered[name] = true;
```

Comment:
@zombieJ Could have used `this.renderFields` instead of the new key, but it is being set true on invoking `getFieldDecorator` itself, so maybe not setting it there may break some behavior.

Reply:
`renderFields` is used to check if user render the field in a render function cause form will remove field if not render anymore. It's correct behavior.

But for element real render seems can only check if ref dom exist in view. If not have best way to check the real dom, I guess reuse `renderFields` for check this may be a relatively viable approach.

Reply:
So shall I update the code to use `renderFields` instead?

Reply:
Seems it's easiest way. yes...

Reply:
@zombieJ  Done